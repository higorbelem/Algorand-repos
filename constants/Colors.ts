const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
