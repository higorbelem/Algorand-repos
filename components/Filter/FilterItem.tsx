import { StyleSheet, TouchableOpacity } from "react-native"

import { Colors } from "@/constants/Colors";

import { Text } from "../Text";
import { filterItemType } from ".";

export type FilterItemProps = filterItemType & {
   selected: boolean;
   onPress: () => void;
};

export const FilterItem = ({label, selected, onPress}: FilterItemProps) => {
    const style = styles(selected);

    return(
        <TouchableOpacity testID="filter-item" style={style.container} activeOpacity={0.7} onPress={onPress}>
            <Text>{label}</Text>
        </TouchableOpacity>
    )
}

const styles = (selected: boolean) => StyleSheet.create({
    container: {
        height: 40,
        paddingHorizontal: 20,
        borderRadius: 20,
        backgroundColor: selected ? Colors.main : '#ffffff22',
        alignItems: 'center',
        justifyContent: 'center'
    }
});