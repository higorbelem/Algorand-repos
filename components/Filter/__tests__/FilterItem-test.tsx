import * as React from 'react';
import { render } from '@testing-library/react-native';

import { Colors } from '@/constants/Colors';

import { FilterItem } from '../FilterItem';

const mockData = {id: 'perawallet', label: 'Pera Wallet'};

describe('FilterItem component', () => {
  it('Renders correctly', () => {
    const tree = render(<FilterItem {...mockData} selected onPress={() => {}}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show correct text', () => {
    const tree = render(<FilterItem {...mockData} selected onPress={() => {}}/>);

    const component = tree.getByText(mockData.label);

    expect(component).toBeTruthy();
  });

  it('Should change background if selected', () => {
    const tree = render(<FilterItem {...mockData} selected onPress={() => {}}/>);

    const component = tree.getByTestId("filter-item");

    expect(component).toHaveStyle({ backgroundColor: Colors.main });
  });

  it('Should change background if not selected', () => {
    const tree = render(<FilterItem {...mockData} selected={false} onPress={() => {}}/>);

    const component = tree.getByTestId("filter-item");

    expect(component).toHaveStyle({ backgroundColor: '#ffffff22' });
  });
});