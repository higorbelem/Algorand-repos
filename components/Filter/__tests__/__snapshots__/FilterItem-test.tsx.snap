// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem component Renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2D2DF1",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
    }
  }
  testID="filter-item"
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
        },
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "400",
        },
      ]
    }
  >
    Pera Wallet
  </Text>
</View>
`;
